### DISCUSSION TITLE
Previous job runs should be overridden by subsequent runs

### DISCUSSION THREAD

**Discussion by parzhitsky** [community/community#25020] [Actions]
<p>There are a couple of types of <code>pull_request</code> event, such as <code>labeled</code> and <code>unlabeled</code>, that does not assume that tree has changed (as opposed to <code>opened</code> or <code>synchronize</code> for example). However, for all the checks, their results are assigned to the latest commit in the pull request. This leads to the situation when different results of the same workflow (invoked several times) get accumulated.</p>
<p>The problem with that is that even for the same workflow, different runs might lead to incompatible outcomes, which does not seem to be addressed.</p>
<p>In my particular case, I’m trying to check proper labels being set on a PR. I’m listening for <code>labeled</code> and <code>unlabeled</code> events. When the labels are invalid, the error from workflow is shown on the latest commit. Then, after the correct labels have been set, the “okay” workflow response is <em>also</em> shown on the latest commit. And I end up with the commit, which is both correct and incorrect.</p>
<p><img src="https://user-images.githubusercontent.com/59528510/181088453-a384cfaf-0d8f-4c78-901b-498433984b86.png" alt="weird_checks.png" data-base62-sha1="gk1Xekf6vuoEbMnXvabU0wp3PXJ" width="508" height="314"><br>
Can I somehow override results of the previous runs on next runs?</p>

**Comment by weide-zhou** [comment]
<p>‘label’&amp;’unlabel’ are two types of pull request event, they doesn’t have actual code commit, the check runs won’t be grouped up. It’s by designed that cannot override the previous check runs.</p>

**Comment by parzhitsky** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> Apologise, didn’t get that. What do you mean by “won’t be grouped up”? On the screenshot, the results of all checks are assigned to a single commit, - I guess, one could say that they are grouped up together. Perhaps, you mean different thing?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/parzhitsky">@parzhitsky</a> ,</p>
<p>If you check type ‘synchronize’ for pull request event, after the 1st commit check run, if you commit another, in the pull request details, only one check run exists, the new check run will override the previous one.</p>
<p>For ‘labels’ &amp; ‘unlabels’, the previous check runs won’t be overidden, which called not grouped.</p>
<p>This is by design.  Sorry for the confusion taken.</p>

**Comment by parzhitsky** [comment]
<p>Looks like currently I have no way to trigger labels’ validation on PRs, other than adding an (empty) commit :frowning:</p>
